.github/workflows/tests.yaml

```diff
@@ -45,3 +46,31 @@ jobs:
     - name: run executables tests
       run: yarn install && yarn test
       working-directory: 'lib/executables/pensieveCreds/'
+
+  compile:
+    name: Compile and upload build artifacts
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Install NodeJS
+      uses: actions/setup-node@v2
+      with:
+        node-version: '16'
+        cache: yarn
+    - name: Install dependencies
+      run: yarn cache clean && yarn install --frozen-lockfile
+      continue-on-error: true # TODO ARSN-97 Remove it when no errors in TS
+    - name: Compile
+      run: yarn build
+      continue-on-error: true # TODO ARSN-97 Remove it when no errors in TS
+    - name: Upload artifacts
```

Comment:
what are we uploading here and why? It looks like we are uploading the built arsenal project, are we then going to import it later or something?

Reply:
We are uploading the built Arsenal package, because it's not really possible to get the build from GitHub, because we're ignoring the build folder in the git repo. Storing the built is there to be able to depend from a commit hash directly.
I don't know the use case. @jbertran

Reply:
Oh right, cause we don't use NPM